bench_pm: Helper application for testing pm behavior

gebart opened the pull request:
The application assists in measuring wake up delays and for measuring power consumption of different low power modes. A logic analyzer or oscilloscope is required for taking any measurements.

This application was used to test the GPIO interrupt optimizations in RIOT-OS/RIOT#8558

Configured out of the box to run on frdm-kw41z, frdm-k22f, samr21-xpro, but it should be easy to add more configurations (look at the top of main.c)

smlng commented on bench_pm/Makefile:
```diff
@@ -0,0 +1,13 @@
+
+RIOTBASE ?= $(CURDIR)/../../RIOT
+QUIET ?= 1
+CFLAGS += -DDEVELHELP
```
use `DEVELHELP=1` instead

smlng commented on bench_pm/main.c:
```diff
@@ -0,0 +1,208 @@
+ *
+ * This test will switch to different low power modes and wait for wake events.
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
```
you might want to change this, or (at least) add yourself?

gebart commented:
addressed comments

gebart commented:
Updated:
 - Cleaned up defines for pins, better general platform support.
 - Added "coma mode" which runs pm_set(0) in a tight loop without any additional wake up sources, to get a good reference for the minimum achievable power consumption.

kaspar030 commented on bench_pm/main.c:
```diff
@@ -0,0 +1,226 @@
+#endif
+
+#if COMA_MODE
+    puts("Coma mode, going to pm_set(0), not coming back");
```
I've had some boards that made it really difficult to re-attach a debugger when going to deep sleep too fast.
Could we maybe guard this, either by a second delay or using something like ```while (gpio_read(BUTTON1)) {}```?

kaspar030 replied:
> some boards that made it really difficult to re-attach a debugger

Like, the only way was to do "while true; do make -C examples/hello-world flash; done" and hammering the reset button, hoping to hop into the short time from reset to CPU stop...
